lib/queue/job.js

```diff
@@ -543,20 +558,22 @@ Job.prototype.state = function (state, fn) {
             .zrem(client.getKey('jobs:' + this.type + ':' + oldState), this.id);
     }
     multi
-        .hset(client.getKey('job:' + this.id), 'state', state)
         .zadd(client.getKey('jobs:' + state), this._priority, this.id)
         .zadd(client.getKey('jobs:' + this.type + ':' + state), this._priority, this.id);
-
     ('inactive' == state) ? multi.lpush(client.getKey(this.type + ':jobs'), 1) : noop();
 
-    multi.exec(function (err, replies) {
-        if(!err) {
-            this._state = state;
-            this.set('updated_at', Date.now());
-            (this._state === 'inactive') ? events.emit(this.id, 'enqueue', this.type) : noop();
-        }
-        return fn( err );
+    this.set('state', state, function(){
```

Comment:
I remember once I put `set state` before `multi.exec` but can't remember if it was producing problems or not... however it may seem more logical to first change job state, hm?